models/eligibility_check.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * An EligibilityCheck is a request from the User to validate his/her DSU (dichiarazione sostitutiva unica). It might be referred also as DSU Request.
+ * An EligibilityCheck is identified by the User's unique identifier (thus at most one EligibilityCheck can exist for a User at a given time).
+ * It can be a case of Success of Failure: in the first case it will also include a list of the family members of the associated DSU
+ */
+
+import * as DocumentDb from "documentdb";
+import { Either, left, right } from "fp-ts/lib/Either";
+import * as DocumentDbUtils from "io-functions-commons/dist/src/utils/documentdb";
+import { DocumentDbModel } from "io-functions-commons/dist/src/utils/documentdb_model";
+import * as t from "io-ts";
+import { pick, tag } from "italia-ts-commons/lib/types";
+import { EligibilityCheck } from "../generated/definitions/EligibilityCheck";
+import { EligibilityCheckFailure } from "../generated/definitions/EligibilityCheckFailure";
+import { EligibilityCheckSuccess } from "../generated/definitions/EligibilityCheckSuccess";
+import { keys } from "../utils/types";
+
+export const ELIGIBILITY_CHECK_COLLECTION_NAME = "eligibility-checks";
+export const ELIGIBILITY_CHECK_MODEL_PK_FIELD = "id";
+
+interface IRetrievedEligibilityCheck {
+  readonly kind: "IRetrievedEligibilityCheck";
+}
+export const RetrievedEligibilityCheck = tag<IRetrievedEligibilityCheck>()(
+  t.intersection([EligibilityCheck, DocumentDbUtils.RetrievedDocument])
+);
+export type RetrievedEligibilityCheck = t.TypeOf<
+  typeof RetrievedEligibilityCheck
+>;
+
+interface INewEligibilityCheckTag {
+  readonly kind: "INewEligibilityCheck";
+}
+export const NewEligibilityCheck = tag<INewEligibilityCheckTag>()(
+  t.intersection([EligibilityCheck, DocumentDbUtils.NewDocument])
+);
+export type NewEligibilityCheck = t.TypeOf<typeof NewEligibilityCheck>;
+
+function toRetrieved(
+  result: DocumentDb.RetrievedDocument
+): RetrievedEligibilityCheck {
+  return {
+    ...result,
+    kind: "IRetrievedEligibilityCheck"
+  } as RetrievedEligibilityCheck;
+}
+
+function toBaseType(o: RetrievedEligibilityCheck): EligibilityCheck {
+  // removes attributes of RetrievedEligibilityCheck which aren't of EligibilityCheck
+  // TODO: try to use EligibilityCheck.encode(o)
```

Comment:
why does not work?

Reply:
since you're doing a cast anyway (inside keys method), you can consider using something like this
https://github.com/pagopa/io-functions-admin/blob/2ec3768cb304d4c1bac91d4fc48065802b792e6d/ExtractUserDataActivity/handler.ts#L530
and cast v to EligibilityCheck

Reply:
> why does not work?
this works
```typescript
EligibilityCheck.encode(o);
```
but this doesn't
```typescript
t.exact(EligibilityCheck).encode(o);
```
The first is actually unhelpful as it doesn't clear extra attributes.